src/context.ts

```diff
@@ -36,13 +36,15 @@ export interface DBOSContext {
 }
 
 export class DBOSContextImpl implements DBOSContext {
-  request: HTTPRequest = {};          // Raw incoming HTTP request.
-  authenticatedUser: string = "";     // The user that has been authenticated
-  authenticatedRoles: string[] = [];  // All roles the user has according to authentication
-  assumedRole: string = "";           // Role in use - that user has and provided authorization to current function
-  workflowUUID: string = "";          // Workflow UUID. Empty for HandlerContexts.
-  executorID: string = "local";       // Executor ID. Gathered from request headers and "local" otherwise
-  readonly logger: DBOSLogger;      // Wrapper around the global logger for this context.
+  request: HTTPRequest = {};						// Raw incoming HTTP request.
+  authenticatedUser: string = "";					// The user that has been authenticated
+  authenticatedRoles: string[] = [];					// All roles the user has according to authentication
+  assumedRole: string = "";						// Role in use - that user has and provided authorization to current function
+  workflowUUID: string = "";						// Workflow UUID. Empty for HandlerContexts.
+  executorID: string = process.env.DBOS__VMID || "local";		// Executor ID. Gathered from the environment and "local" otherwise
+  applicationVersion: string = process.env.DBOS__APPVERSION || "";	// Application version. Gathered from the environment and empty otherwise
+  applicationID: string = process.env.DBOS__APPID || "";		// Application ID. Gathered from the environment and empty otherwise
```

Comment:
Set these by default in the context

Reply:
Here we set them empty by default, but set to "XXX_NOT_DEFINED" in traces.ts.